docs/docs/features/nats.md

```diff
@@ -0,0 +1,231 @@
+﻿---
+sidebar_position: 3
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import helloWorldModuleComponent from '!!raw-loader!./data/helloworld.module.yaml';
+import natsMessagesFormat from './data/ApiGearNatsv0.1.pdf';
+
+# NATS
+
+:::caution
+This is an experimental feature. It contains the smallest working set of functionalities to adapt the generated interface for use with the NATS protocol.
+It doesn't include any security, the error handling is minimal and it is not production ready. It doesn't use the JetStream nor the Streaming features. Please also check issues on github for this template.
+:::
+
+:::note
+A [CNATS](https://nats-io.github.io/nats.c/) library is used to implement NATS in this library.
+:::
+
+This feature provides NATS protocol adapters into your project, and also show that an existing protocol can be adapted for sharing your data in your ecosystem. When going through this document you may notice this implementation contains general client/server adapters in `📂hello-world/apigear/nats`
+and an interface specific part generated from templates for each interface in `📂hello-world/cpp_hello_world/modules/io_world/generated/nats`.
+
+This feature provides a _client_ and _service_ adapter for your interfaces for the NATS protocol. It allows you to connect different applications in the same or different technologies (check all of our [templates](/docs/sdk/intro)).
+
+- Use an _Nats client_ instead of your interface implementation to be able to receive data from remote service.
+- Use an _Nats server adapter_ to expose your interface implementation as a remote service. You can expose same service for different technologies (olink, mqtt, other adapted protocol).
+
+:::tip
+The NATS server is not provided with the template. To be able to run apigear client and service (both serve as NATS clients) you need to run a server of your choice, which is accessible for both client and service.
+:::
+
+:::caution
+In this library we use STL classes in public interfaces.
+We assume that this library is built from source (configuration provided by generated `CMakeLists`).
```

Comment:
We assume that this library is built from source using cmake

Reply:
I'd rather keep it in a form it is in, we use also conan- true it does use cmake but I am not sure about the wording 🤔